### PR TITLE
change par[i] to pi in streaming

### PR DESCRIPTION
**Pull request by darioizzo**
small change in the name of the parameters in the streaming operator .... clearly name clashes are no more likely if the user uses p1, p2 etc.. for variable names .....

* advantages: future integration in simpy and algebraic manipulators may be easier 
* advantages: the subs method feels more natural
* disatvantage: the API symmetry is broken between streaming operator and the internal variable storage for the parameters which is still called par[i]

### CODE REVIEW

**Comment by bluescarni** [comment]
@darioizzo can you fix the test as well?

File: src/param.cpp [hunk 1]
```diff
@@ -83,9 +83,13 @@ bool operator!=(const param &p0, const param &p1)
     return !(p0 == p1);
 }
 
-expression subs(const param &p, const std::unordered_map<std::string, expression> &)
+expression subs(const param &p, const std::unordered_map<std::string, expression> &smap)
 {
-    return expression{p};
+    if (auto it = smap.find("p"+std::to_string(p.idx())); it == smap.end()) {
```
Comment (bluescarni) [inline src/param.cpp:88]: Can you use ``_format`` here as well, instead of ``to_string()``?

Comment (darioizzo) [reply src/param.cpp]: now outdated, but format would not compile?

Comment (bluescarni) [reply src/param.cpp]: probably forgot the ``using literal`` thing I guess.

**Pull request merged**